pkg/cluster/provider.go

```diff
@@ -105,6 +106,39 @@ func (p *Provider) Delete(name, explicitKubeconfigPath string) error {
 	return internaldelete.Cluster(p.logger, p.ic(name), explicitKubeconfigPath)
 }
 
+// Deletes one or more clusters created from list of cluster names and using Delete function.
+func (p *Provider) DeleteSome(clusters []string, explicitKubeconfigPath string) error {
+	for _, cluster := range clusters {
+		err := internaldelete.Cluster(p.logger, p.ic(cluster), explicitKubeconfigPath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete cluster %q", cluster)
+		}
+	}
+
+	return nil
+}
+
+// Deletes all clusters created by getting list of cluster names and using Delete function.
+func (p *Provider) DeleteAll(explicitKubeconfigPath string) error {
```

Comment:
I don't think we should put this footgun in the API, developers of other software should not be doing this.

an end user confirming it in the CLI is one thing, but some test rig importing the go library almost definitely should not be deleting all clusters, only the ones it created (with some known prefix or something...)

Reply:
this is also just doing a List() call in the CLI if `--all` is specified, and then the same code path as the non-all case.